azuredevops/internal/service/servicehook/publishers.go

```diff
@@ -0,0 +1,158 @@
+package servicehook
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var eventTypeMap = map[string]map[string]string{
+	"pipelines": {
+		"stage_state_changed": "ms.vss-pipelines.stage-state-changed-event",
+		"run_state_changed":   "ms.vss-pipelines.run-state-changed-event",
+	},
+	"distributedtasks": {},
+	"rm":               {},
+}
+
+var publisherResourceVersionMap = map[string]string{
+	"pipelines":        "5.1-preview.1",
+	"distributedtasks": "1.0-preview.1",
+	"rm":               "3.0-preview.1",
+}
+
+func genPublisherSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"publisher": {
+			Type:     schema.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"name": {
+						Type:         schema.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice([]string{"pipelines", "distributedtasks", "rm"}, false),
+					},
+					"stage_state_changed": {
```

Comment:
This common schema should only available to `pipeline` type publisher not other types even they have same parameters.